cloud-agent/service/server/src/main/scala/org/hyperledger/identus/pollux/credentialdefinition/controller/CredentialDefinitionControllerImpl.scala

```diff
@@ -44,28 +64,54 @@ class CredentialDefinitionControllerImpl(service: CredentialDefinitionService, m
     service
       .getByGUID(guid)
       .map(
-        fromDomain(_)
+        CredentialDefinitionResponse
+          .fromDomain(_)
           .withSelf(rc.request.uri.toString)
       )
   }
 
+  override def getCredentialDefinitionByGuidDidUrl(
+      baseUrlServiceName: String,
+      guid: UUID
+  )(implicit rc: RequestContext): IO[ErrorResponse, PrismEnvelopeResponse] = {
+
+    val res = for {
+      cd <- service.getByGUID(guid, ResourceResolutionMethod.did)
+      response <- ZIO
+        .fromEither(CredentialDefinitionDidUrlResponse.asPrismEnvelopeResponse(cd, baseUrlServiceName))
+        .mapError(couldNotParseCredDefResponse)
+
+    } yield response
+
+    res
+  }
```

Comment:
Would it be better to use a `for-comprehension` without `val res` in the beginning and `res` in the end?

Reply:
its a stylistic choise, I like this because its easier to debug later, and see the response type by hovering over.